feat(docs): Api examples

abhifetch opened the pull request.

abhifetch committed:
Fix: Changing permisson for LLM API page

abhifetch committed:
fix(docs): updating permission for LLM API page

abhifetch committed:
fix(docs): updating permission for LLM API page

abhifetch committed:
feat(docs): Chat API example

abhifetch committed:
fix: minor edits

abhifetch committed:
Merge branch 'master' into APIExamples

abhifetch committed:
fix: minor changes

abhifetch committed:
minor changes

devjsc commented on pages/apis/ai-engine/chat_api_example.mdx:
```diff
@@ -0,0 +1,628 @@
+
+    - Python version greater than 3.9 and less than 3.11.
+    - The requests library installed. You can install it using pip install requests.
+    - An API token (fauna) from Agentverse, which will be used for authentication.
```
How do we get this, link to the file that describes it.

devjsc commented on pages/apis/ai-engine/chat_api_example.mdx:
```diff
@@ -0,0 +1,628 @@
+- Before you begin, ensure you have the following:
+
+    - Python version greater than 3.9 and less than 3.11.
+    - The requests library installed. You can install it using pip install requests.
```
pip install requests

devjsc commented on pages/apis/ai-engine/chat_api_example.mdx:
```diff
@@ -0,0 +1,628 @@
+
+- Steps to get API Token:
+
+    1. Login to agentverse.
```
agentverse should link to agentverse.ai

devjsc commented on pages/apis/ai-engine/chat_api_example.mdx:
```diff
@@ -0,0 +1,628 @@
+    2. Right click on chrome browser and select Inspect
+    3. Go to the application tab and look for cookies.
+    4. Select Agenverse.ai and select fauna cookie.
+    5. Copy the fauna’s value and this is your token key.
```
oh god @ejfitzgerald what is the solution to this?

devjsc commented on pages/apis/ai-engine/chat_api_example.mdx:
````diff
@@ -0,0 +1,628 @@
+
+    ``` python
+    # Define function to normalize questions asked by AI-engine
+    def normalize_question(question):
````
might be nice to have type here, so 

`def normalize_question(question : str):`

devjsc commented on pages/apis/ai-engine/chat_api_example.mdx:
````diff
@@ -0,0 +1,628 @@
+        return question_normalized
+    ```
+
+    This function converts questions from a JSON response, identified by the key 'text', into lowercase and removes any punctuation. 
````
Which function? the next one?

devjsc replied:
I can't see `.text` am i missing something? 😅

devjsc commented on pages/apis/ai-engine/chat_api_example.mdx:
````diff
@@ -0,0 +1,628 @@
+        response = requests.post(f"https://agentverse.ai/v1beta1/engine/chat/sessions/{session_id}/submit", json=data, headers={"Authorization": token})
+    ```
+
+    This function is designed to send user's choice back to AI-engine through Chat API, specially in case of user getting options from multiple
````
Would be great to add an action here such as: 

To understand the flow from Ab to B to C, click here ↗️

devjsc replied:
@FelixNicolaeBucsa

devjsc commented on pages/apis/ai-engine/chat_api_example.mdx:
````diff
@@ -0,0 +1,628 @@
+        response = requests.post(f"https://agentverse.ai/v1beta1/engine/chat/sessions/{session_id}/submit", json=data, headers={"Authorization": token})
+    ```
+
+    This function is designed to send a message from the user to the chat session. This is used to submit user input as part of the chat flow.
````
```suggestion
    This function to sends a message from the user to the chat session. This is used to submit user input as part of the chat flow.
```

devjsc commented on pages/apis/ai-engine/chat_api_example.mdx:
````diff
@@ -0,0 +1,628 @@
+        return pattern.match(key)
+    ```
+
+    This function checks whether the given key is in UUID (Universal Unique Identifier) format.The function uses a regular expression pattern to match the 
````
```suggestion
    `is_uuid` accepts a key argument, which then verified is the key given is in the correct UUID format. 
```

devjsc commented on pages/apis/ai-engine/chat_api_example.mdx:
````diff
@@ -0,0 +1,628 @@
+    ```
+
+    This function checks whether the given key is in UUID (Universal Unique Identifier) format.The function uses a regular expression pattern to match the 
+    standard UUID format, which consists of 32 hexadecimal characters split into five groups separated by hyphens (8-4-4-4-12).
````
```suggestion

```

devjsc commented on pages/apis/ai-engine/chat_api_example.mdx:
```diff
@@ -0,0 +1,628 @@
+    This function checks whether the given key is in UUID (Universal Unique Identifier) format.The function uses a regular expression pattern to match the 
+    standard UUID format, which consists of 32 hexadecimal characters split into five groups separated by hyphens (8-4-4-4-12).
+
+    The function takes the 'key' parameter and returns match object if key is UUID else it returns None.
```
```suggestion

```

devjsc commented on pages/apis/ai-engine/chat_api_example.mdx:
````diff
@@ -0,0 +1,628 @@
+    print("Session stopped:", response.json())
+    ```
+
+    This function is designed to stop the ongoing deltaV chat session.
````
```suggestion
    `stop_session` posts a request to the chat api to stop the session passed.
```

abhifetch committed:
Update pages/apis/ai-engine/chat_api_example.mdx

Co-authored-by: Joshua Croft <32483134+devjsc@users.noreply.github.com>

abhifetch committed:
Update pages/apis/ai-engine/chat_api_example.mdx

Co-authored-by: Joshua Croft <32483134+devjsc@users.noreply.github.com>

abhifetch committed:
Update pages/apis/ai-engine/chat_api_example.mdx

Co-authored-by: Joshua Croft <32483134+devjsc@users.noreply.github.com>

abhifetch committed:
Update chat_api_example.mdx

minor changes

abhifetch committed:
Update pages/apis/ai-engine/chat_api_example.mdx

Co-authored-by: Joshua Croft <32483134+devjsc@users.noreply.github.com>

abhifetch committed:
Update chat_api_example.mdx

adding Powering connections and smart operations in DeltaV to get better understanding

abhifetch committed:
minor changes

abhifetch committed:
Including example and guides

abhifetch committed:
adding tags

abhifetch committed:
minor changes

abhifetch committed:
minor changes

FelixNicolaeBucsa commented on pages/examples.mdx:
```diff
@@ -92,6 +92,14 @@ import { GuideBox } from "../components/feature-guide-tabs";
 
+    <GuideBox
+        content={{
+            title: "AI-Engine Chat API",
```
AI Engine

FelixNicolaeBucsa commented on pages/examples/chat-api.mdx:
```diff
@@ -0,0 +1,248 @@
+
+### Guide
+
+- [AI-Engine Chat API guide ↗️](/guides/apis/chat_api_guide)
```
AI Engine

FelixNicolaeBucsa commented on pages/examples/chat-api.mdx:
```diff
@@ -0,0 +1,248 @@
+    import time
+    import json
+
+    # define function to normalize questions asked by AI-engine
```
**D**efine

FelixNicolaeBucsa commented on pages/guides.mdx:
```diff
@@ -679,4 +679,17 @@ import { GuideBox } from "../components/feature-guide-tabs";
+<Grid3>
+    <GuideBox
+        content={{
+            title: "AI-Engine Chat API",
```
AI Engine

FelixNicolaeBucsa commented on pages/guides/_meta.json:
```diff
@@ -4,5 +4,6 @@
   "fetch-network": "Fetch Network",
-  "agent-courses": "Courses"
+  "agent-courses": "Courses",
+  "apis": "API"
```
APIs

FelixNicolaeBucsa commented on pages/guides/apis/_meta.json:
```diff
@@ -0,0 +1,6 @@
+{
+  "chat_api_guide": {
+    "title": "AI-Engine Chat API guide",
```
AI Engine

FelixNicolaeBucsa committed:
fix: conflicts

FelixNicolaeBucsa committed:
fix: conflicts

abhifetch committed:
minor fixes

abhifetch committed:
AI Engine changes

abhifetch committed:
adding guide link to chat API documentation

abhifetch committed:
minor changes
